cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/LineageService.java

```diff
@@ -49,10 +50,11 @@
       }
     };
 
-  private final LineageDataset lineageDataset;
+  private final LineageStore lineageStore;
 
-  public LineageService(LineageDataset lineageDataset) {
-    this.lineageDataset = lineageDataset;
+  @Inject
+  public LineageService(LineageStore lineageStore) {
```

Comment:
For injected constructor, better make it package private to private direct instantiation (exception for unit-test in the same package).